### PR TITLE
Create ci.yml

### PR DESCRIPTION
**Pull request by QartemisT**

### CODE REVIEW

**Comment by BreakBB** [comment]
Hey @QartemisT looks like you read our minds! I was planning to add CI to Questie this week 😄 Thanks for taking the time! I will merge this as soon as #2585 is merged 👍🏻

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+        with:
+          fetch-depth: 100
+
+      - name: Install and run Luacheck
+        uses: nebularg/actions-luacheck@v1
```
Comment (BreakBB) [inline .github/workflows/ci.yml:23]: There is also a `luacheck@v2` version available already which we should use.

Moreover will this also run luacheck and not just install it? 🤔 For my other addon I added:

```yml
      - uses: leafo/gh-actions-lua@v8.0.0
        with:
          luaVersion: "5.1"

      - uses: leafo/gh-actions-luarocks@v4.0.0

      - name: build
        run: |
          luarocks install luacheck
          luacheck .
```

to install lua + luarocks and run the luacheck command.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
```
Comment (BreakBB) [inline .github/workflows/ci.yml:18]: Oh and there is also a v2 of this action.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (QartemisT) [reply .github/workflows/ci.yml:23]: Yeh it does both installing and running.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 100
+
+      - name: Install and run Luacheck
+        uses: nebularg/actions-luacheck@v2
+        with:
+          args: "--no-color -q"
```
Comment (BreakBB) [inline .github/workflows/ci.yml:25]:
```suggestion
          args: "Database Localization --no-color -q"
```

**Review by BreakBB — APPROVED** [review]

**Pull request merged**